src/hooks/useRestyle.ts

```diff
@@ -39,28 +34,39 @@ const useRestyle = <
   TRestyleProps extends Record<string, any>,
   TProps extends TRestyleProps & {style?: StyleProp<RNStyle>}
 >(
-  restyleFunctions: (
-    | RestyleFunctionContainer<TProps, Theme>
-    | RestyleFunctionContainer<TProps, Theme>[])[],
+  composedRestyleFunction: {
+    buildStyle: <TInputProps extends TProps>(
+      props: TInputProps,
+      {
+        theme,
+        dimensions,
+      }: {
+        theme: Theme;
+        dimensions: Dimensions;
+      },
+    ) => ViewStyle | TextStyle | ImageStyle;
+    properties: (keyof TProps)[];
+    propertiesMap: Record<keyof TProps, boolean>;
+  },
   props: TProps,
 ) => {
   const theme = useTheme<Theme>();
 
   const dimensions = useDimensions();
 
   const restyled = useMemo(() => {
-    const composedRestyleFunction = composeRestyleFunctions(restyleFunctions);
-    const style = composedRestyleFunction.buildStyle(props, {
+    const {cleanProps, restyleProps} = filterRestyleProps(
+      props,
+      composedRestyleFunction.propertiesMap,
+    );
+    const style = composedRestyleFunction.buildStyle(restyleProps, {
```

Comment:
We were calling `composedRestyleFunction.buildStyle` with all the props of the components, but we only need to call it with the props that belong to `restyle`.